client/lib/wp/sync-handler/cache-invalidation.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+import { getLocalForage } from 'lib/localforage';
+
+/**
+ * Module variables
+ */
+const localforage = getLocalForage();
+const debug = debugFactory( 'calypso:sync-handler:cache' );
+const RECORDS_LIST_KEY = 'records-list';
+
+// set record lifetime to 2 days
+const DAY_IN_HOURS = 24;
+const HOUR_IN_MS = 3600000;
+export const LIFETIME = 2 * DAY_IN_HOURS * HOUR_IN_MS;
+
+export class CacheInvalidation {
+	getAll( callback = () => {} ) {
+		return localforage.getItem( RECORDS_LIST_KEY, callback );
+	}
+
+	/**
+	 * Add the given `key` into the cache-invalidation object
+	 * adding at the same a marktime (now).
+	 * If the pair key-mark already exists it will be updated.
+	 *
+	 * @param {String} key - record key
+	 * @return {Promise} promise
+	 */
+	addItem( key, fn = () => {} ) {
+		return this.filterByKey( key, ( err, records = [] ) => {
+			debug( 'adding %o', key );
+
+			// add the fresh item into history list
+			records.unshift( { key, mark: new Date() } );
```

Comment:
IndexedDB does support storing date object, but localforage may use other drivers that don't support this. I suggest changing this to the following, and updating the related usages.
```javascript
mark: Date.now()
```

Additionally, Chrome dev tools will incorrectly show the date object as an empty object:
![screen shot 2016-02-22 at 12 37 50 pm](https://cloud.githubusercontent.com/assets/1270189/13231876/1ce5dcf4-d961-11e5-8ff7-56031f4f6656.png)